platform/engineering/design-docs/2020-05-26-content-validation.md

```diff
@@ -0,0 +1,378 @@
+
+## High Level Design
+
+Once the [build separation work](https://github.com/department-of-veterans-affairs/va.gov-team/issues/2719) is complete, there will be separate content and application builds. These distinct builds will prevent failures in one from blocking deployments to the other.
+
+The Metalsmith static content build will be extracted to the `content-build` repo and deleted from the `vets-website` repo and deployment pipeline. Webpack will handle the application build for the `vets-website` repo, and Metalsmith will handle the static content build for the `content-build` repo.
+
+The validation of static content will take place in a scheduled job that runs every workday, instead of happening as part of the full deploy. Since the broken link check happens during the Metalsmith build, and the accessibility check relies on the output of the content build, the scheduled job will run a script to produce a build that only has the steps necessary for content validation. That build will only be used for reporting broken links and accessibility errors, and will have no impact on deploys. Those reports will be delivered to the #cms-team channel via existing Slack integrations.
+
+For content writers, this approach preserves their ability to quickly draft, publish, and deploy content. And the improved reporting will make it easier for them to pinpoint errors. The downside of this approach is that content writers still won't have a proactive way to fully validate their content.
+
+For the Frontend Tools team, this approach will reduce noise in the application build logs, and eliminate a source of failed application builds. Eliminating a source of failed application builds means the Frontend Tools team will have to spend less time manually triaging failed application builds, which will give the team more bandwidth to work on other priorities.
+
+For anybody trying to ship code for the VA.gov front end, this approach will eliminate a source of failed application builds, which should allow them to ship code faster.
+
+For VA.gov users, this approach will continue allowing users to quickly see updated content. The downside of this approach for VA.gov users is that it won't eliminate existing opportunities for users to see accessibility errors. Fortunately, the improved error reporting should allow content writers to easily identify, remedy, and deploy valid content, which should minimize the user's chance of encountering invalid content.
+
+In summary, this approach will have the following impact for the following groups:
+
+- **Content Writers**: Improve error reporting while maintaining current convenience
+- **Frontend Tools team**: Reduce build noise, failure rate, and number of builds requiring manual intervention
+- **VA.gov users**: Maintain existing time to see updated content and risk of seeing accessibility errors
+- **Anybody trying to ship code for the VA.gov front end**: Reduce failed deploys which should allow them to ship faster
```

Comment:
given our conversations about broken link checking still blocking deploys after all... what builds will we no longer be failing?

Reply:
With this approach, the application build will no longer fail due to accessibility errors. 

Phrased another way:
**Current**: a11y content errors break the application build, but not the content build
**Proposed**: a11y content errors will not break the application build, and continue not breaking the content build